### ISSUE TITLE
detect if machines did not finish provisioning

### ISSUE THREAD

**Issue by kwmonroe** [juju-solutions/cloud-weather-report#46]
In this job, we see that machine-4 never finished provisioning:

http://juju-ci.vapour.ws/job/cwr-test/361/consoleFull

```
"4":

    agent-state: pending

    dns-name: 104.197.160.101

    instance-id: juju-79765edc-b5ba-4cf8-8264-714d8b8807c7-machine-4

    instance-state: RUNNING

    series: trusty
```

We currently raise this as a TimeoutError.  This is not entirely true because no amount of timeout would have made this machine available (if it's pending after an hour, it's gonna be pending forever).

Let's analyze juju status (which we capture at the end of the consoleFull log), and if we have machines in 'pending', mark this as a "provisioning" or "cloud" failure, not a timeout.

**Comment by kwmonroe** [comment]
Also, if we detect this kind of failure, let's not use a red "x".  The red "x" makes people think the bundle failed in some way, when it was a cloud issue.

Seman will check other html codes to see what might be more appropriate than a red/green icon.

**Comment by kwmonroe** [comment]
Just another data point.  Here's another 'pending' machine that would be good to show as "provisioning issue":

http://juju-ci.vapour.ws/job/cwr-test/370/consoleFull

```
"7":

    agent-state: pending

    dns-name: juju-cwr-default-azure-jdyxxojjkk.cloudapp.net

    instance-id: juju-cwr-default-azure-jdyxxojjkk-jujukz5w4ntdnkhgwrsata3kbly5ht6taba7rl9s5limh8k0th

    instance-state: ReadyRole

    series: trusty

    hardware: arch=amd64 cpu-cores=2 mem=7168M root-disk=130048M
```

I can't tell why this machine didn't come up, but I think the appropriate visual for this kind of failure is some warning code (not the red 'x').

**Comment by johnsca** [comment]
For the "no result" item, Antonio suggested a yellow triangle: `&#x25B2` = &#x25B2;

**Comment by seman** [comment]
Resolved by #50
